aiida_quantumespresso/workflows/protocols/utils.py

```diff
@@ -70,6 +70,8 @@ def recursive_merge(left, right):
     """
     import collections
 
+    right = right.copy()
```

Comment:
I was going to say that we needed a `deepcopy` here to also copy nested dictionaries, but since this function is called recursively for nested dictionaries, copy should work. Maybe we can add a comment of this here.